src/jit/flowgraph.cpp

```diff
@@ -1044,10 +1044,27 @@ flowList* Compiler::fgAddRefPred(BasicBlock* block,
 
     assert(!fgCheapPredsValid);
 
-    flowList* flow = fgGetPredForBlock(block, blockPred);
+    flowList* flow;
 
-    if (flow)
+    // Keep the predecessor list in lowest to highest bbNum order. This allows us to discover the loops in
+    // optFindNaturalLoops from innermost to outermost.
+    //
+    // TODO-Throughput: This search is quadratic if you have many jumps to the same target. We need to either
```

Comment:
Correct, what's quadratic is adding N predecessor edges to a block, requiring searching the predecessor list for each one.